webpack/scenes/ContentViews/Table/tableDataGenerator.js

```diff
@@ -105,23 +105,15 @@ const buildDetailDropdowns = (id, rowIndex, openColumn) => {
   return detailDropdowns;
 };
 
-const tableDataGenerator = (results, rowMapping) => {
-  const updatedRowMap = {};
-  const contentViews = results || [];
-  const columns = buildColumns();
+const buildRowsAndMapping = (contentViews, newRowMapping) => {
+  const updatedRowMap = { ...newRowMapping };
   const rows = [];
-  const contentViewIds = contentViews.map(cv => cv.id);
-
-  // Only keep the relevant rows to keep the table status check accurate
-  Object.entries(rowMapping).forEach(([rowId, value]) => {
-    if (contentViewIds.includes(value.id)) updatedRowMap[rowId] = value;
-  });
 
   contentViews.forEach((contentView) => {
     const { id } = contentView;
     const rowIndex = rows.length;
     const needsUpdate = !Object.keys(updatedRowMap).find(i => updatedRowMap[i].id === id) ||
-                        !Object.keys(updatedRowMap[rowIndex]).includes('expandedColumn');
+                        !Object.keys(updatedRowMap[rowIndex] || {}).includes('expandedColumn');
```

Comment:
If `updatedRowMap[rowIndex]` were undefined or null, it would throw a TypeError (I checked in my console.) So why `|| {}` ?